### PR TITLE
fix wrong identification of python 3.10 as python 3.1

### PR DESCRIPTION
**Pull request by basilrabi**
QGIS fails to build if the python installed in the system is higher than [3.9](https://download.copr.fedorainfracloud.org/results/@python/python3.10/fedora-rawhide-x86_64/02127449-qgis/builder-live.log.gz). This fixes it by using regex instead of hard-coded substring.

### CODE REVIEW

File: cmake/FindLibPython.py [hunk 1]
```diff
@@ -32,9 +32,10 @@
 
 import sys
 import distutils.sysconfig
+from re import sub
 
 print("exec_prefix:%s" % sys.exec_prefix)
-print("short_version:%s" % sys.version[:3])
+print("short_version:%s" % sub(r'(^\d+\.\d+).*', r'\1', sys.version.split()[0]))
```
Comment (hroncok) [inline cmake/FindLibPython.py:38]: What about this?

```suggestion
print("short_version:%d.%d" % sys.version_info[:2])
```

**Comment by elpaso** [comment]
Can you move this PR to master instead of 3.18?

File: cmake/FindLibPython.py (hunk 1 above)
Comment (basilrabi) [reply cmake/FindLibPython.py]: Yes this is better.

**Comment by basilrabi** [comment]
> Can you move this PR to master instead of 3.18?

This was my first though but `cmake/FindLibPython.py` does not exist in master.

**Comment by elpaso** [comment]
> > Can you move this PR to master instead of 3.18?
> 
> This was my first thought but `cmake/FindLibPython.py` does not exist in master.

You are right, it was reworked with https://github.com/qgis/QGIS/pull/42497

**Review by 3nids — APPROVED** [review]

**Pull request merged**